### DISCUSSION TITLE
How to set custom Action button for a per row basis?

### DISCUSSION THREAD

**Discussion by samik-os** [thedevdojo/voyager#5260] [Q&A]
I am trying to implement an **Action** for my Controller **where the action will only appear for rows where the data has a certain value**. In this case, I want this Action button **to show only for Orders that have NOT been delivered**. Please observe the `shouldActionDisplayOnDataType` function of my Action class below, as the method I tried makes the action button disappear for ALL rows.

```
use App\Order;

class OrderManageAction extends AbstractAction
{
    public function getTitle()
    {
        return 'Manage';
    }

    public function getIcon()
    {
        return 'voyager-tools';
    }

    public function getPolicy()
    {
        return 'manage';
    }

    public function getAttributes()
    {
        return [
            'class' => 'btn btn-sm btn-success pull-left',
        ];
    }

    public function getDefaultRoute()
    {
        return route('order.manage', $this->data->{$this->data->getKeyName()});
    }

    public function shouldActionDisplayOnDataType()
    {
        // this line makes the action button DISAPPEAR for ALL rows
        return ($this->dataType->slug == 'orders' && $this->data->order_state_id != Order::STATE_DELIVERED);

        // this line makes the action button APPEAR for ALL rows
        // return $this->dataType->slug == 'orders';
    }
}

```

Apparently, Voyager calls this check method only once before drawing the UI. Is there a way to do this so that I can perform this check method for every row called?

**Comment by Cyrille37** [comment]
You can do it in `getAttributes()` that 's called on each row.

```
 if( $this->data->slug == 'lorem-ipsum-post')
   return ['class' => 'hidden'];
 else
   return ['class' => 'btn btn-sm btn-warning pull-right view'];
```

**Comment by emptynick** [comment]
I think we could include a simple method `shouldBeDisplayedForRow()` (or similar) which by default returns true.

Edit: https://github.com/the-control-group/voyager/pull/5355
